### PR TITLE
feat(TDP-4390): configure csv escape character

### PR DESCRIPTION
**Pull request by ncomont**
**Link to the JIRA issue**
https://jira.talendforge.org/browse/TDP-4087
https://jira.talendforge.org/browse/TDP-4389
https://jira.talendforge.org/browse/TDP-4390

**Please check if the PR fulfills these requirements**
- [x] The commit(s) message(s) follows our [guidelines](https://github.com/talend/tools/blob/master/tools-root-github/CONTRIBUTING.md#commit-message-format)
- [x] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)
- [ ] The code coverage on new code is > 75 % for backend and > 95% for frontend
- [x] The new code does not introduce new technical issues (sonar / eslint)
- [x] Functional tests have been performed
- [ ] Docker configuration files for config-std or config-cloud profiles are impacted

**Please check the browsers you've tested on**
- [x] Chrome, Firefox, Safari, Edge, IE11
- [ ] No, that's bad, this PR should not be merged !
- [ ] No, and no need to (backend changes only)

### CODE REVIEW

**Comment by build-talend-tdp** [comment]
Pull request test result: [failure](https://ci-tdp.datapwn.com/job/data-prep_pull-request_master/880/) [:x:](https://ci-tdp.datapwn.com/job/data-prep_pull-request_master/880/)

**Comment by build-talend-tdp** [comment]
Pull request test result: [failure](https://ci-tdp.datapwn.com/job/data-prep_pull-request_master/886/) [:x:](https://ci-tdp.datapwn.com/job/data-prep_pull-request_master/886/)

File: dataprep-transformation/src/main/java/org/talend/dataprep/transformation/format/CSVWriter.java [hunk 1]
```diff
@@ -70,19 +87,17 @@ public CSVWriter(final OutputStream output) {
...
-            if (separatorParameter == null || StringUtils.isEmpty(separatorParameter) || separatorParameter.length() > 1) {
-                this.separator = String.valueOf(DEFAULT_SEPARATOR).charAt(0);
-            } else {
-                this.separator = separatorParameter.charAt(0);
-            }
+
+            this.separator = this.getParameterValue(parameters, SEPARATOR_PARAM_NAME, DEFAULT_SEPARATOR);
+            this.escapeCharacter = this.getParameterValue(parameters, ESCAPE_CHARACTER_PARAM_NAME, DEFAULT_ESCAPE_CHARACTER);
+
             bufferFile = File.createTempFile("csvWriter", ".csv");
```
Comment (fabballe) [inline dataprep-transformation/src/main/java/org/talend/dataprep/transformation/format/CSVWriter.java]: You should add a timestamp at the end of the file name in order to be sure that concurrency export will not affect the other export

Comment (fhuaulme) [reply dataprep-transformation/src/main/java/org/talend/dataprep/transformation/format/CSVWriter.java]: No need to, File.createTempFile() already generates a file with a random number in file name.

**Comment by build-talend-tdp** [comment]
Pull request test result: [failure](https://ci-tdp.datapwn.com/job/data-prep_pull-request_master/893/) [:x:](https://ci-tdp.datapwn.com/job/data-prep_pull-request_master/893/)

**Comment by build-talend-tdp** [comment]
Pull request test result: [failure](https://ci-tdp.datapwn.com/job/data-prep_pull-request_master/894/) [:x:](https://ci-tdp.datapwn.com/job/data-prep_pull-request_master/894/)

**Comment by build-talend-tdp** [comment]
Pull request test result: [failure](https://ci-tdp.datapwn.com/job/data-prep_pull-request_master/897/) [:x:](https://ci-tdp.datapwn.com/job/data-prep_pull-request_master/897/)

**Comment by build-talend-tdp** [comment]
Pull request test result: [failure](https://ci-tdp.datapwn.com/job/data-prep_pull-request_master/898/) [:x:](https://ci-tdp.datapwn.com/job/data-prep_pull-request_master/898/)

**Comment by build-talend-tdp** [comment]
Pull request test result: [failure](https://ci-tdp.datapwn.com/job/data-prep_pull-request_master/899/) [:x:](https://ci-tdp.datapwn.com/job/data-prep_pull-request_master/899/)

**Comment by build-talend-tdp** [comment]
Pull request test result: [failure](https://ci-tdp.datapwn.com/job/data-prep_pull-request_master/901/) [:x:](https://ci-tdp.datapwn.com/job/data-prep_pull-request_master/901/)

**Comment by build-talend-tdp** [comment]
Pull request test result: [failure](https://ci-tdp.datapwn.com/job/data-prep_pull-request_master/904/) [:x:](https://ci-tdp.datapwn.com/job/data-prep_pull-request_master/904/)

**Comment by build-talend-tdp** [comment]
Pull request test result: [failure](https://ci-tdp.datapwn.com/job/data-prep_pull-request_master/905/) [:x:](https://ci-tdp.datapwn.com/job/data-prep_pull-request_master/905/)

**Comment by build-talend-tdp** [comment]
Pull request test result: [failure](https://ci-tdp.datapwn.com/job/data-prep_pull-request_master/907/) [:x:](https://ci-tdp.datapwn.com/job/data-prep_pull-request_master/907/)

**Comment by build-talend-tdp** [comment]
Pull request test result: [failure](https://ci-tdp.datapwn.com/job/data-prep_pull-request_master/908/) [:x:](https://ci-tdp.datapwn.com/job/data-prep_pull-request_master/908/)

**Comment by build-talend-tdp** [comment]
Pull request test result: [failure](https://ci-tdp.datapwn.com/job/data-prep_pull-request_master/909/) [:x:](https://ci-tdp.datapwn.com/job/data-prep_pull-request_master/909/)

**Comment by build-talend-tdp** [comment]
Pull request test result: [failure](https://ci-tdp.datapwn.com/job/data-prep_pull-request_master/910/) [:x:](https://ci-tdp.datapwn.com/job/data-prep_pull-request_master/910/)

**Comment by build-talend-tdp** [comment]
Pull request test result: [failure](https://ci-tdp.datapwn.com/job/data-prep_pull-request_master/911/) [:x:](https://ci-tdp.datapwn.com/job/data-prep_pull-request_master/911/)

**Comment by build-talend-tdp** [comment]
Pull request test result: [failure](https://ci-tdp.datapwn.com/job/data-prep_pull-request_master/913/) [:x:](https://ci-tdp.datapwn.com/job/data-prep_pull-request_master/913/)

**Comment by build-talend-tdp** [comment]
Pull request test result: [failure](https://ci-tdp.datapwn.com/job/data-prep_pull-request_master/914/) [:x:](https://ci-tdp.datapwn.com/job/data-prep_pull-request_master/914/)

**Comment by build-talend-tdp** [comment]
Pull request test result: [failure](https://ci-tdp.datapwn.com/job/data-prep_pull-request_master/920/) [:x:](https://ci-tdp.datapwn.com/job/data-prep_pull-request_master/920/)

**Comment by build-talend-tdp** [comment]
Pull request test result: [failure](https://ci-tdp.datapwn.com/job/data-prep_pull-request_master/923/) [:x:](https://ci-tdp.datapwn.com/job/data-prep_pull-request_master/923/)

**Comment by build-talend-tdp** [comment]
Pull request test result: [failure](https://ci-tdp.datapwn.com/job/data-prep_pull-request_master/924/) [:x:](https://ci-tdp.datapwn.com/job/data-prep_pull-request_master/924/)

**Comment by build-talend-tdp** [comment]
Pull request test result: [failure](https://ci-tdp.datapwn.com/job/data-prep_pull-request_master/925/) [:x:](https://ci-tdp.datapwn.com/job/data-prep_pull-request_master/925/)

**Comment by build-talend-tdp** [comment]
Pull request test result: [failure](https://ci-tdp.datapwn.com/job/data-prep_pull-request_master/926/) [:x:](https://ci-tdp.datapwn.com/job/data-prep_pull-request_master/926/)

**Comment by build-talend-tdp** [comment]
Pull request test result: [failure](https://ci-tdp.datapwn.com/job/data-prep_pull-request_master/928/) [:x:](https://ci-tdp.datapwn.com/job/data-prep_pull-request_master/928/)

**Comment by build-talend-tdp** [comment]
Pull request test result: [failure](https://ci-tdp.datapwn.com/job/data-prep_pull-request_master/935/) [:x:](https://ci-tdp.datapwn.com/job/data-prep_pull-request_master/935/)

**Comment by build-talend-tdp** [comment]
Pull request test result: [failure](https://ci-tdp.datapwn.com/job/data-prep_pull-request_master/938/) [:x:](https://ci-tdp.datapwn.com/job/data-prep_pull-request_master/938/)

**Comment by build-talend-tdp** [comment]
Pull request test result: [failure](https://ci-tdp.datapwn.com/job/data-prep_pull-request_master/939/) [:x:](https://ci-tdp.datapwn.com/job/data-prep_pull-request_master/939/)

**Comment by build-talend-tdp** [comment]
Pull request test result: [failure](https://ci-tdp.datapwn.com/job/data-prep_pull-request_master/940/) [:x:](https://ci-tdp.datapwn.com/job/data-prep_pull-request_master/940/)

**Comment by build-talend-tdp** [comment]
Pull request test result: [failure](https://ci-tdp.datapwn.com/job/data-prep_pull-request_master/941/) [:x:](https://ci-tdp.datapwn.com/job/data-prep_pull-request_master/941/)

**Comment by build-talend-tdp** [comment]
Pull request test result: [failure](https://ci-tdp.datapwn.com/job/data-prep_pull-request_master/942/) [:x:](https://ci-tdp.datapwn.com/job/data-prep_pull-request_master/942/)

**Comment by build-talend-tdp** [comment]
Pull request test result: [failure](https://ci-tdp.datapwn.com/job/data-prep_pull-request_master/943/) [:x:](https://ci-tdp.datapwn.com/job/data-prep_pull-request_master/943/)

**Comment by build-talend-tdp** [comment]
Pull request test result: [failure](https://ci-tdp.datapwn.com/job/data-prep_pull-request_master/944/) [:x:](https://ci-tdp.datapwn.com/job/data-prep_pull-request_master/944/)

**Comment by build-talend-tdp** [comment]
Pull request test result: [failure](https://ci-tdp.datapwn.com/job/data-prep_pull-request_master/945/) [:x:](https://ci-tdp.datapwn.com/job/data-prep_pull-request_master/945/)

**Comment by build-talend-tdp** [comment]
Pull request test result: [failure](https://ci-tdp.datapwn.com/job/data-prep_pull-request_master/946/) [:x:](https://ci-tdp.datapwn.com/job/data-prep_pull-request_master/946/)

**Comment by build-talend-tdp** [comment]
Pull request test result: [failure](https://ci-tdp.datapwn.com/job/data-prep_pull-request_master/947/) [:x:](https://ci-tdp.datapwn.com/job/data-prep_pull-request_master/947/)

**Comment by build-talend-tdp** [comment]
Pull request test result: [failure](https://ci-tdp.datapwn.com/job/data-prep_pull-request_master/948/) [:x:](https://ci-tdp.datapwn.com/job/data-prep_pull-request_master/948/)

**Comment by build-talend-tdp** [comment]
Pull request test result: [success](https://ci-tdp.datapwn.com/job/data-prep_pull-request_master/954/) [:star:](https://ci-tdp.datapwn.com/job/data-prep_pull-request_master/954/)

**Comment by build-talend-tdp** [comment]
Pull request test result: [failure](https://ci-tdp.datapwn.com/job/data-prep_pull-request_master/958/) [:x:](https://ci-tdp.datapwn.com/job/data-prep_pull-request_master/958/)

File: dataprep-backend-common/src/main/java/org/talend/dataprep/parameters/SelectParameter.java [hunk 2]
```diff
@@ -46,18 +51,22 @@
...
         super(name, ParameterType.SELECT, defaultValue, implicit, canBeBlank);
-        this.isRadio = isRadio;
+        setRadio(radio);
         setItems(items);
         setMultiple(multiple);
     }
 
-    public boolean isRadio() {
-        return isRadio;
+    public boolean getRadio() {
```
Comment (yacarof) [inline dataprep-backend-common/src/main/java/org/talend/dataprep/parameters/SelectParameter.java]: A boolean getter isn't supposed to start by "is" ? like : 
boolean isRadio();

**Comment by build-talend-tdp** [comment]
Pull request test result: [failure](https://ci-tdp.datapwn.com/job/data-prep_pull-request_master/961/) [:x:](https://ci-tdp.datapwn.com/job/data-prep_pull-request_master/961/)

**Comment by build-talend-tdp** [comment]
Pull request test result: [failure](https://ci-tdp.datapwn.com/job/data-prep_pull-request_master/975/) [:x:](https://ci-tdp.datapwn.com/job/data-prep_pull-request_master/975/)

**Comment by build-talend-tdp** [comment]
Pull request test result: [success](https://ci-tdp.datapwn.com/job/data-prep_pull-request_master/977/) [:star:](https://ci-tdp.datapwn.com/job/data-prep_pull-request_master/977/)

**Pull request merged**